hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequest.java

```diff
@@ -250,13 +250,12 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
               omBucketInfo.getDefaultReplicationConfig(),
               ozoneManager);
 
-      validateEncryptionKeyInfo(omBucketInfo, keyArgs);
-
       omKeyInfo = prepareKeyInfo(omMetadataManager, keyArgs, dbKeyInfo,
           keyArgs.getDataSize(), locations, getFileEncryptionInfo(keyArgs),
           ozoneManager.getPrefixManager(), omBucketInfo, pathInfo, trxnLogIndex,
           ozoneManager.getObjectIdFromTxId(trxnLogIndex),
           ozoneManager.isRatisEnabled(), repConfig);
+      validateEncryptionKeyInfo(omBucketInfo, keyArgs);
```

Comment:
moving this validation after prepareKeyInfo() so that in the case of invalid upload id, prepareKeyInfo() throws the proper exception. Otherwise, validateEncryptionKeyInfo() would throw an OMException with INTERNAL_ERROR code that crashes OM.